openedx/features/edly/utils.py

```diff
@@ -0,0 +1,218 @@
+import logging
+import re
+
+from django.conf import settings
+
+from contentstore.utils import get_lms_link_for_item, is_currently_visible_to_students
+from courseware.courses import get_course_by_id
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+from openedx.core.djangoapps.theming.helpers import get_current_site
+from openedx.features.edly.tasks import send_bulk_mail_to_students
+from student.models import CourseEnrollment
+from xmodule.contentstore.content import StaticContent
+
+log = logging.getLogger(__name__)
+
+COURSE_OUTLINE_CATEGORIES = ['vertical', 'sequential', 'chapter']
+
+
+def notify_students_about_xblock_changes(xblock, publish, old_content):
+    """
+    This function is responsible for calling the related function by checking the xblock category.
+
+    Arguments:
+        xblock: Block of courses which has been published.
+        publish: Param to check if xblock is going to be published.
+        old_content: Old data of xblock before updating.
+    """
+    if (publish == 'make_public' and xblock.category in COURSE_OUTLINE_CATEGORIES
+            and is_currently_visible_to_students(xblock)):
+        _handle_section_publish(xblock)
+    elif xblock.category == 'course_info' and xblock.location.block_id == 'handouts':
+        _handle_handout_changes(xblock, old_content)
+
+
+def get_email_params(xblock):
+    """
+    Generates the email params for any changes in course
+
+    Arguments:
+        xblock: xblock which is modified/created
+
+    Returns:
+        Dict containing the data for email.
+    """
+    email_params = {}
+    course = get_course_by_id(xblock.location.course_key)
+    email_params['site'] = get_current_site()
+    email_params['course_url'] = _get_course_url(xblock.location.course_key)
+    email_params['course_name'] = course.display_name_with_default
+    email_params['display_name'] = xblock.display_name
+    email_params['platform_name'] = settings.PLATFORM_NAME
+    email_params['site_name'] = configuration_helpers.get_value(
+        'SITE_NAME',
+        settings.SITE_NAME
+    )
+    return email_params
+
+
+def _get_course_url(course_key):
+    return '{}/courses/{}'.format(settings.LMS_ROOT_URL, course_key)
```

Comment:
do we need to make this url on our own, try using:
```
reverse('course_root', args=[self.course_run_key])
```
OR
```
reverse('courseware', args=[course_id])
```

Reply:
This both `url_name` is for lms. We can't use them in cms.